core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ReactorSinkProviderBasedSink.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.strategy;
+
+import org.mule.runtime.api.lifecycle.Disposable;
+import org.mule.runtime.core.api.construct.BackPressureReason;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.processor.Sink;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+
+/**
+ * {@link Sink} implementation that uses a {@link Flux} for each thread that dispatches events to it.
```

Comment:
actually this class does not encapsulate neither the cache logic nor the stream per thread logic. This simply is a sink that delegates the retrieval of a sink to a ReactorSinkProvider so I would change the jdoc.